#167190408 Search functionality

Lundii opened the pull request:
#### What does this PR do?
It helps a user search and filter an article
#### Description of Task to be completed?
Added search functionality
#### How should this be manually tested?
Type a search word on the search bar
#### Any background context you want to provide?
[End to End testing video](https://drive.google.com/file/d/1HfRE5juVg3wIAA18QeGDSAMf_B_o3dSf/view?usp=sharing)
#### What are the relevant pivotal tracker stories?
[#167190408] (https://www.pivotaltracker.com/story/show/167190408)
#### Screenshots (if appropriate)
<img width="1440" alt="Screenshot 2019-09-07 at 7 30 15 AM" src="https://user-images.githubusercontent.com/37886814/64470871-c33ac980-d141-11e9-9241-45c89c09b57a.png">
<img width="1433" alt="Screenshot 2019-09-07 at 7 29 41 AM" src="https://user-images.githubusercontent.com/37886814/64470881-d8aff380-d141-11e9-95a7-6e4f15204626.png">

#### Questions:
No

Lundii committed:
ft(frontend): Search functionality

- design the search page
- write tests

[Finishes #168107415]

Lundii committed:
Code cleanup

topseySuave commented on src/reducers/searchReducer.js:
```diff
@@ -0,0 +1,83 @@
+      };
+    }
+
+    case 'UPDATE_PAGENUMBER': {
```
This actions should not be defined here, cause it has already been defined in the action types. 
which should be referenced here instead like so `actionType. UPDATE_PAGENUMBER` they are constants.

topseySuave replied:
Please make the necessary changes for all the defined action in this file and other files.

Lundii replied:
That's true. I have made the necessary changes. Thanks

topseySuave commented on src/views/SearchPage/SearchBody.scss:
```diff
@@ -0,0 +1,19 @@
+
+.padding {
+  margin: 20px;
+}
```
Add a new line here and other files that have the same warning.

Lundii replied:
Done

topseySuave commented on src/views/SearchPage/SearchPage.scss:
```diff
@@ -0,0 +1,21 @@
+    grid-area: footer;
+    margin: 20px;
+  }
+}
```
Here.

Lundii replied:
Done

Lundii committed:
ft(frontend): Search Functionality
 - design the search page with react
- connect it to redux
- write test

[Finishes #167190408]

chuxmykel commented on .eslintrc:
```diff
@@ -40,7 +43,7 @@
     "no-param-reassign": 0,
     "comma-dangle": 2,
-    "no-console": 1,
+    "no-console": 0,
```
This should be set to 1 so as to remind us to remove console statements before pushing.

Lundii replied:
Alright. I made it zero when my server was failing because of console.log. I will change it now. Thanks for the observation.

Lundii replied:
> This should be set to 1 so as to remind us to remove console statements before pushing.

Done

chuxmykel reviewed (approved):
Nice work @Lundii. Looks good to me.

Hadeneekeh commented on src/views/HomePage/index.js:
```diff
@@ -1,3 +1,88 @@
-import HomePage from './HomePage';
```
Great job @Lundii!

I can see that you brought in the content of Homepage.js into this file. I am confused, and I would like to know why you made the changes.

Lundii replied:
It shouldn't be there at all. I will remove it now. Thanks for the observations.

Lundii replied:
> Great job @Lundii!
> 
> I can see that you brought in the content of Homepage.js into this file. I am confused, and I would like to know why you made the changes.

Done

Lundii committed:
Minor fixes

Hadeneekeh reviewed (approved):
LGTM!

encodedBicoding reviewed (approved):
LGTM!

chiboycalix commented on src/lib/AuthStore.js:
```diff
@@ -22,6 +22,7 @@ export default class AuthStore {
 
   static getToken() {
-    return localStorage.getItem(this.TOKEN_NAME);
+    // return localStorage.getItem(this.TOKEN_NAME);
```
you can get rid of this console.log if it's not useful
